src/lib/components/ProductForm.svelte

```diff
@@ -299,6 +299,41 @@
 	function isNumber(value: string) {
 		return !isNaN(Number(value)) && value.trim() !== '';
 	}
+	function moveVariation(fromIndex: number, toIndex: number): void {
+		if (product.variations) {
+			if (
+				fromIndex < 0 ||
+				toIndex < 0 ||
+				fromIndex >= product.variations.length ||
+				toIndex >= product.variations.length
+			) {
+				return;
+			}
+			const newVariations = [...product.variations];
+			const [movedItem] = newVariations.splice(fromIndex, 1);
+			newVariations.splice(toIndex, 0, movedItem);
+			product.variations = newVariations;
+		}
+	}
+	function duplicateVariation(
+		variation: { name: string; value: string; price?: number },
+		index: number,
+		newVariation: boolean
+	) {
+		if (newVariation) {
+			variationLabelsNames.push(variation.name);
+			variationLabelsValues.push(variation.value);
+		} else {
+			const newVariationValue = crypto.randomUUID();
+			if (product.variationLabels) {
+				product.variationLabels.values[variation.name][newVariationValue] =
+					product.variationLabels.values[variation.name][variation.value];
+			}
+			product.variations?.splice(index + 1, 0, { ...variation, value: newVariationValue });
+		}
```

Comment:
I don't understand what's the relationship between `variationLabelsValues, variationLabelsNames` and `product.variationLabels`.

if anything I'd expect the two former to be lists of the keys and values of the later one. But this code reveals that they don't even have the same shape. What's the UUID seemingly being used as a key?